transform.ml

```diff
@@ -1,7 +1,7 @@
 open Instr
 
 (* types of transformations *)
-type transform_instructions = instruction_stream -> instruction_stream option
+type transform_instructions = analysis_input -> analysis_input option
```

Comment:
Are there concrete cases where we want to remove some of the formal parameters? If not, would it be better to return `instructions`?